### PR TITLE
Provide Additional Context for Chunk Options in APIs

### PR DESCRIPTION
**Pull request by yuhongsun96**

### CODE REVIEW

File: backend/danswer/search/pipeline.py [hunk 1]
```diff
@@ -48,15 +77,148 @@ def __init__(
...
+        # If chunks have been merged, the LLM filter flow no longer applies
+        # as the indices no longer match. Can be implemented later as needed
+        self.ran_merge_chunk = False
 
         # generator state
         self._postprocessing_generator: Generator[
             list[InferenceChunk] | list[str], None, None
         ] | None = None
 
+    def _combine_chunks(self, post_rerank: bool) -> list[InferenceSection]:
```
Comment (yuhongsun96) [inline backend/danswer/search/pipeline.py:95]: Please think through this section, should be correct hopefully

File: backend/danswer/search/pipeline.py [hunk 2]
```diff
@@ -48,15 +77,148 @@ def __init__(
...
+                )
+            )
+
+        # list of list of inference chunks where the inner list needs to be combined for content
+        list_inference_chunks = run_functions_tuples_in_parallel(
+            functions_with_args, allow_failures=False
+        )
+
+        for chunk_range in reverse_map.values():
+            inf_chunks = list_inference_chunks.pop(0)
```
Comment (Weves) [inline backend/danswer/search/pipeline.py]: Can we pop from the back? I believe popping from the front is a bit slower: https://stackoverflow.com/a/34633242/22737043

File: backend/danswer/search/pipeline.py [hunk 3]
```diff
@@ -48,15 +77,148 @@ def __init__(
...
+                            ),
+                        )
+                    )
+
+            list_inference_chunks = run_functions_tuples_in_parallel(
+                functions_with_args, allow_failures=False
+            )
+
+            for chunk in unique_chunks:
+                inf_chunks = list_inference_chunks.pop(0)
```
Comment (Weves) [inline backend/danswer/search/pipeline.py]: same comment here

File: backend/danswer/search/pipeline.py [hunk 4]
```diff
@@ -114,47 +276,70 @@ def retrieved_docs(self) -> list[InferenceChunk]:
...
+        self._relevant_chunk_indices = [
             ind
             for ind, chunk in enumerate(reranked_docs)
             if chunk.unique_id in relevant_chunk_ids
         ]
-        return self._relevant_chunk_indicies
+        return self._relevant_chunk_indices
 
     @property
     def chunk_relevance_list(self) -> list[bool]:
```
Comment (Weves) [inline backend/danswer/search/pipeline.py:336]: maybe split this into two functions, `chunk_relevance_list` and `section_relevance_list`

File: backend/danswer/search/pipeline.py (hunk 3 above)
Comment (yuhongsun96) [reply backend/danswer/search/pipeline.py]: The ordering must be maintained for this to be correct, we could do everything from the back potentially but, I don't want to complicate it, I'll just use indices instead.

But also the scale of this is fairly small, shouldn't matter no?

**Pull request merged**